core/rawdb/accessors_trie.go

```diff
@@ -311,6 +311,28 @@ func ReadStateScheme(db ethdb.Reader) string {
 	return HashScheme
 }
 
+// ReadStateSchemeByStateDB reads the state scheme of persistent state from state disk db, or none
+// if the state is not present in database
+func ReadStateSchemeByStateDB(db, statediskdb ethdb.Reader) string {
```

Comment:
Simply pass `db` as a single parameter and use `db.StateStore` in it.

Reply:
refactored